src/ClpSolverInterface.jl

```diff
@@ -237,3 +237,195 @@ end
 getrawsolver(m::ClpMathProgModel) = m.inner
 
 end
+
+module ClpMOISolverInterface
+using Clp.ClpCInterface
+
+# importall MathProgBase.SolverInterface
+
+using MathOptInterface
+const MOI = MathOptInterface
+
+export ClpMOISolver
+
+# Solver
+
+immutable ClpMOISolver <: MOI.AbstractSolver
+    options
+end
+ClpMOISolver(;kwargs...) = ClpMOISolver(kwargs)
+
+# Solver instance
+
+const VR = MOI.VariableReference
+const CR = MOI.ConstraintReference
+
+type ClpMOIModel <: MOI.AbstractSolverInstance
+    inner::ClpModel
+    solveroptions::ClpSolve
+    nvariables::UInt64
+    vartolbref::Vector{CR}
+    vartoubref::Vector{CR}
+    ubreftovar::Dict{CR, Int32}
+    lbreftovar::Dict{CR, Int32}
+    nconstraints::UInt64 # does take into account bounds
```

Comment:
It s not really needed. nconstraints can be also deduced as (2 * nvariables + nrows).